### PR TITLE
Trigger history node validation & trimming logic within write path

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Trigger history node validation & trimming logic for the following APIs (write path)
  * CreateWorkflowExecution: no auto fix since this API is creating a brand new workflow
  * ConflictResolveWorkflowExecution: auto fix for reset workflow & current workflow
  * UpdateWorkflowExecution: auto fix for update workflow
  

**Why?**
Add additional protection, similar to #2366 (read path)
For details see #1449

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/persistence/execution_manager.go [hunk 1]
```diff
@@ -937,6 +977,40 @@ func (m *executionManagerImpl) Close() {
...
+	workflowID string,
+	runID string,
+) {
+	response, err := m.GetWorkflowExecution(&GetWorkflowExecutionRequest{
+		ShardID:     shardID,
+		NamespaceID: namespaceID,
+		WorkflowID:  workflowID,
+		RunID:       runID,
+	})
+	if err != nil {
```
Comment (paulnpdev) [inline common/persistence/execution_manager.go:992]: I understand why this only needs to be best-efforts, but still it should be VERY rare that GetWorkflowExecution fails here.  Did you consider logging a warning here?  If this lookup fails it seems likely it's a bug somewhere.

Maybe (eventually) it's even a candidate for a panic in dev? ... not suggesting we do that now.

**Review by paulnpdev — APPROVED** [review]

**Comment by paulnpdev** [comment]
BTW, I don't think the description of the PR matches what it does very well.  Maybe reword that a bit?

**Review by yiminc — APPROVED** [review]

File: common/persistence/execution_manager.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/execution_manager.go:992]: >Maybe (eventually) it's even a candidate for a panic in dev?

this error can happen if DB is temporarily unavailable, so error can be expected (compared to in mem state check)

added logging here, with namespace & workflow & run.

**Comment by wxing1292** [comment]
> BTW, I don't think the description of the PR matches what it does very well. Maybe reword that a bit?

sure, updated the title, PTAL

**Comment by paulnpdev** [comment]
much better, thank you.
